vsts_cd_manager/continuous_delivery_manager.py

```diff
@@ -155,20 +155,37 @@ def _verify_vsts_parameters(self, cd_account, source_repository):
         if source_repository.type in ['Github', 'ExternalGit'] and not cd_account:
             raise RuntimeError('You must provide a value for cd-account since your repo-url is not a Team Services repository.')
 
-    def _get_build_configuration(self, app_type, working_directory):
+    def _get_build_configuration(self, app_type_details, working_directory):
+
+        accepted_app_type = ['AspNetWap', 'AspNetCore', 'NodeJS', 'PHP', 'Python']
+        accepted_nodejs_task_runner = ['None', 'Gulp', 'Grunt']
+        accepted_python_framework = ['Bottle', 'Django', 'Flask']
+        accepted_python_version = ['Python 2.7.12 x64', 'Python 2.7.12 x86', 'Python 2.7.13 x64', 'Python 2.7.13 x86', 'Python 3.5.3 x64', 'Python 3.5.3 x86', 'Python 3.6.0 x64', 'Python 3.6.0 x86', 'Python 3.6.2 x64', 'Python 3.6.1 x86']
+        
         build_configuration = None
-        if app_type == 'AspNetWap':
-            build_configuration = BuildConfiguration(app_type, working_directory)
-        elif app_type == 'AspNetCore':
-            build_configuration = BuildConfiguration(app_type, working_directory)
-        elif app_type == 'PHP':
+        app_type = app_type_details.cd_app_type
+        if (app_type == 'AspNetWap') or (app_type == 'AspNetCore') or (app_type == 'PHP') :
             build_configuration = BuildConfiguration(app_type, working_directory)
-        elif app_type == 'NodeJSWithGulp':
-            build_configuration = BuildConfiguration('NodeJS', working_directory, 'Gulp')
-        elif app_type == 'NodeJSWithGrunt':
-            build_configuration = BuildConfiguration('NodeJS', working_directory, 'Grunt')
+        elif app_type == 'NodeJS' :
```

Comment:
Another way to handle the compat is to change the Azure CLI so that it is locked to a major version instead of getting latest. Then you can safely move the major version of the cd manager up to 2 and then change the Azure CLI to consume 2. Developers using an older version of the source will still get the 1.0 version and shouldn't have any compat issues.